techniques/fileConfiguration/fileManagement/4.1/manageFilesAndFolders.st

```diff
@@ -0,0 +1,317 @@
+			perms        => mg("${extended_modes_${index}}${file[${index}][mode]}", "${file[${index}][group]}"),
+			classes      => rudder_common_classes("manageFilesAndFolders_perms_${index}_RudderUniqueID"),
+			ifvarclass   => "manageFilesAndFolders_check_permissions_${index}.manageFilesAndFolders_isrecursive_${index}.manageFilesAndFolders_groupset_${index}.!manageFilesAndFolders_userset_${index}";
+
+		# This will not delete the parent
+		"${file[${index}][path]}" 
+			delete       => tidy,
+			file_select  => date_pattern("${file[${index}][ttl]}", "${file[${index}][pattern]}"),
+			ifvarclass   => "manageFilesAndFolders_clean_${index}.manageFilesAndFolders_deletion_single_recursive_${index}",
+			depth_search => recurse("0"),
+			classes      => rudder_common_classes("manageFilesAndFolders_delete_${index}_RudderUniqueID");
+
+		# This will not delete the parent
+		"${file[${index}][path]}" 
+			delete       => tidy,
+			file_select  => date_pattern("${file[${index}][ttl]}", "${file[${index}][pattern]}"),
+			depth_search => recurse("inf"),
+			ifvarclass   => "manageFilesAndFolders_clean_${index}.(manageFilesAndFolders_deletion_full_recursive_${index}|manageFilesAndFolders_deletion_exterminate_${index})",
+			classes      => rudder_common_classes("manageFilesAndFolders_delete_${index}_RudderUniqueID");
+
+		# This WILL delete the parent
+		"${file[${index}][path]}" 
+			delete       => tidy,
+			file_select  => date_pattern("${file[${index}][ttl]}", "${file[${index}][pattern]}"),
+			ifvarclass   => "manageFilesAndFolders_clean_${index}.(manageFilesAndFolders_deletion_simple_${index}|manageFilesAndFolders_deletion_exterminate_${index})",
+			classes      => rudder_common_classes("manageFilesAndFolders_delete_${index}_RudderUniqueID");
+
+  methods:
+
+    pass3::
+
+			# Reporting for creation
+			## Reporting for directory creation
+      "any" usebundle => rudder_common_report("manageFilesAndFolders", "result_success", "${file[${index}][uuid]}", "File", "${file[${index}][path]}", "The folder ${file[${index}][path]} already exists"),
+				ifvarclass => "manageFolders_${index}_RudderUniqueID_kept.!manageFolders_${index}_RudderUniqueID_repaired.manageFilesAndFolders_isdir_${index}.!manageFilesAndFolders_isfile_${index}.!manageFilesAndFolders_islink_${index}";
+
+      "any" usebundle => rudder_common_report("manageFilesAndFolders", "result_repaired", "${file[${index}][uuid]}", "File", "${file[${index}][path]}", "The folder ${file[${index}][path]} was successfully created"),
+				ifvarclass => "manageFolders_${index}_RudderUniqueID_repaired";
+
+      "any" usebundle => rudder_common_report("manageFilesAndFolders", "result_error", "${file[${index}][uuid]}", "File", "${file[${index}][path]}", "The folder ${file[${index}][path]} could not be created"),
+				ifvarclass => "manageFolders_${index}_RudderUniqueID_error|manageFolders_${index}_RudderUniqueID_kept.(manageFilesAndFolders_isfile_${index}|manageFilesAndFolders_islink_${index})";
```

Comment:
this is not easy to spot, but if it's a folder we want to create, but it's a file, and promise is kept, then it's dubious and an error